src/utils/chat.ts

```diff
@@ -1,6 +1,74 @@
-import { BaseChatMessage } from "store/actions/Chat";
+import firebase from "firebase/app";
+
+import { WithId, withId } from "utils/id";
+
+import {
+  BaseChatMessage,
+  ChatMessage,
+  MessageToDisplay,
+  PreviewChatMessageToDisplay,
+  PreviewChatMessage,
+  PrivateChatMessage,
+} from "types/chat";
+import { User } from "types/User";
 
 export const chatSort: (a: BaseChatMessage, b: BaseChatMessage) => number = (
   a: BaseChatMessage,
   b: BaseChatMessage
 ) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf());
+
+export interface GetMessageToDisplayProps<T> {
+  message: T;
+  usersById: Record<string, User>;
```

Comment:
Typing issues RE: `Record` and users potentially not existing for every single possible string (described more elsewhere in this review)